ui/apps/platform/src/services/CollectionsService.ts

```diff
@@ -0,0 +1,234 @@
+};
+
+/**
+ * Fetch a paginated list of Collection objects
+ */
+export function listCollections(
+    searchFilter: SearchFilter,
+    sortOption: ApiSortOption,
+    page: number,
+    pageSize: number
+): CancellableRequest<CollectionResponse[]> {
+    const params = getListQueryParams(searchFilter, sortOption, page, pageSize);
+    return makeCancellableAxiosRequest((signal) =>
+        axios
+            .get<{
+                collections: CollectionResponse[];
+            }>(`${collectionsBaseUrl}?${params}`, { signal })
+            .then((response) => response.data.collections)
+    );
+}
+
+/**
+ * Fetch the number of collections
+ */
+export function getCollectionCount(searchFilter: SearchFilter): CancellableRequest<number> {
+    const query = getRequestQueryStringForSearchFilter(searchFilter);
+    return makeCancellableAxiosRequest((signal) =>
+        axios
+            .get<{ count: number }>(`${collectionsCountUrl}?query=${query}`, { signal })
+            .then((response) => response.data.count)
+    );
+}
+
+export type ResolvedCollectionResponse = {
+    collection: CollectionResponse;
+    deployments: ListDeployment[];
+};
+
+/**
+ * Fetch a single collection by id
+ *
+ * @param id
+ *      The collection ID
+ * @param options.withMatches
+ *      When true, returns the list of deployments that match the collection
+ *      rules, otherwise returns an empty array.
+ */
+export function getCollection(
+    id: string,
+    options: { withMatches: boolean } = { withMatches: false }
+): CancellableRequest<ResolvedCollectionResponse> {
+    const params = qs.stringify(options);
+    return makeCancellableAxiosRequest((signal) =>
+        axios
+            .get<ResolvedCollectionResponse>(`${collectionsBaseUrl}/${id}?${params}`, { signal })
+            .then((response) => response.data)
+    );
+}
+
+export type GetCollectionSelectorsResponse = {
+    selectors: SelectorField[];
+};
+
+/**
+ * Create a new collection
+ *
+ * @param collection
+ *      The collection object details to be created
+ * @returns
+ *      The created collection object, with ID
+ */
+export function createCollection(
+    collection: CollectionRequest
+): CancellableRequest<CollectionResponse> {
+    return makeCancellableAxiosRequest((signal) =>
+        axios
+            .post<CollectionResponse>(collectionsBaseUrl, collection, { signal })
+            .then((response) => response.data)
+    );
+}
+
+/**
+ * Updates an existing collection object.
+ *
+ * @param id
+ *      The ID of the collection to update
+ * @param collection
+ *      The new collection object details.
+ * @returns
+ *      The updated collection object
+ *
+ */
+export function updateCollection(
+    id: string,
+    collection: CollectionRequest
+): CancellableRequest<CollectionResponse> {
+    return makeCancellableAxiosRequest((signal) =>
+        axios
+            .post<CollectionResponse>(`${collectionsBaseUrl}/${id}`, collection, { signal })
```

Comment:
Maybe I missed this in the design doc reviews, but why is "update" functionality a `POST`, instead of a `PUT` or a `PATCH` ?

Reply:
I had noticed this too in the backend design doc, since I initially had a `PUT` here. I brought it up during the Tuesday collections sync and Oscar mentioned that he thought that it was intentional but was going to follow up with the team to find out, along with some other questions.

Do we want to hold off merging for now, or change it later if that backend confirms it should be a different method?

Reply:
@dvail @keyallis 
Unless there's a obstacle to using the more semantic PUT or PATCH verbs for updates, I think we should get both front and back-end teams aligned on that now.

Reply:
For this, since the entire payload is being sent on each request to update all fields of the collection I'd say we go with `PUT`. @keyallis I'll go ahead and update this PR if that works on your end.